h2quic/client.go

```diff
@@ -178,7 +178,7 @@ func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 		requestedGzip = true
 	}
 	// TODO: add support for trailers
-	endStream := !hasBody
+	endStream := !hasBody && req.Method != http.MethodConnect
```

Comment:
In some cases, (e.g. my POC codes), I cannot set req.Body filed before RoundTrip it.
Doing a preRead should solve this, but it maybe bring other side effects.